src/br/com/cafegoxtoso/domain/Cafe.java

```diff
@@ -16,12 +14,21 @@ public enum Cafes {
     private final String nome;
     private final double valor;
 
-    Cafes(int codigo, String nome, double valor) {
+    Cafe(int codigo, String nome, double valor) {
         this.codigo = codigo;
         this.nome = nome;
         this.valor = valor;
     }
 
+    public static Cafe cafePorCodigo(int codigoCafe){
+        for (Cafe cafe : values()) {
+            if (cafe.getCodigo() == codigoCafe){
+                return cafe;
+            }
+        }
+        return null;
+    }
```

Comment:
public static Optional<Cafe> cafePorCodigo(int codigoCafe){
        return Stream.of(values()).filter(cafe -> cafe.getCodigo() == codigoCafe).findFirst();
    }